src/shared/components/code-editor/hooks/useSwaggerDefinitions.ts

```diff
@@ -0,0 +1,31 @@
+import { queryOptions, useQuery } from '@tanstack/react-query';
+import { commonFetch } from '../../../../k8s';
+import { OpenAPISchemaDefinitions } from '../types';
+
+const SWAGGER_QUERY_KEY = 'swagger';
+
+const fetchSwaggerDefinitions = async (): Promise<OpenAPISchemaDefinitions> => {
+  try {
+    const response = await commonFetch('/openapi/v2');
```

Comment:
we're fetching the OpenAPI v2 definitions to configure the Monaco editor. This allows the YAML editor to show property descriptions when hovering over it:

![Screenshot 2025-04-23 at 16 23 40](https://github.com/user-attachments/assets/6bb418bf-a524-4fee-86cb-42e098162923)